src/disclosure-manager/DisclosureContainer.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useContext, useRef } from 'react';
+import PropTypes from 'prop-types';
+
+import { ApplicationLoadingOverlayProvider } from '../application-loading-overlay';
+import { NavigationPromptCheckpoint, navigationPromptResolutionOptionsShape } from '../navigation-prompt';
+import ApplicationErrorBoundary from '../application-error-boundary';
+
+import DisclosureManagerContext from './DisclosureManagerContext';
+
+const propTypes = {
+  /**
+   * The components to render within the context of the DisclosureContainer.
+   */
+  children: PropTypes.node,
+  /**
+   * The Object (or function that returns an Object) that specifies the messages
+   * used to prompt the user when disclosure dismissal occurs when pending state
+   * is present.
+   */
+  navigationPromptResolutionOptions: navigationPromptResolutionOptionsShape,
+};
+
+const DisclosureContainer = ({ children, navigationPromptResolutionOptions }) => {
+  const disclosureManager = useContext(DisclosureManagerContext);
+  const promptCheckpointRef = useRef();
+  const registeredPromptsRef = useRef();
+
+  useEffect(() => {
+    disclosureManager.registerDismissCheck(() => new Promise((resolve, reject) => {
+      if (!promptCheckpointRef.current) {
```

Comment:
I don't think the `promptCheckpointRef` is needed. The NavigationPromptCheckpoint manages the list of registered props and will resolve from resolvePrompts:
https://github.com/cerner/terra-framework/blob/master/packages/terra-navigation-prompt/src/NavigationPromptCheckpoint.jsx#L121

Reply:
Good call, that guy has been hiding out from an earlier implementation

Reply:
So just to clarify, the promptCheckpointRef is required for me to be able to call resolvePrompts at all. What *isn't* required is the registeredPromptsRef from the onPromptChange below, which I have removed.